scripts/dep_update.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash -xe
+# Run pre and run hooks
+# Detect whether the repo has been modified
+# Create pr if required
+# Run post hook
+run_hook(){
+  if [[ -x "$1" ]]
+  then
+    $1 || exit 1
+  else
+    if [[ "$1" =~ .*/run ]]
+    then
+      echo "Run hook $1 not found."
+      exit 1
+    fi
+  fi
+}
+cd ${WORKSPACE}/repo
+run_hook "gating/update_dependencies/pre"
+run_hook "gating/update_dependencies/run"
+if [[ -z "$(git status -s)" ]]
+then
+  echo "Repo is clean, prep script made no changes to be comitted"
+else
+  # split URL into array seperated by '/'s
+  IFS='/ ' read -r -a url_split <<< "$URL"
+  # owner/org is whatever is between the second and third / in the url
+  owner="${url_split[3]}"
+  # repo is anything after owner/
+  repo=${URL#https://github.com/${owner}/}
+  ssh_url="git@github.com:${owner}/${repo}"
+  pr_branch="${BRANCH}_dep_update_$(date +%H%M_%d%m%Y)"
+  echo "Repo is dirty, proposing changes"
+  git commit -a -m "${pr_branch}"
```

Comment:
The commit message on the test job is:
```
RE-197_dep_update_1716_27092017
```
I'd like to see this be a bit more human readable. While everyone would eventually understand what that meant, it doesn't cost us to be a bit more descriptive and then the commit will never require explanation. For example,
```
RE-XXX Update {branch} dependencies

This change updates the branch dependencies with the repository
modifications made by running the Jenkins hook in
`gating/update_dependencies`.
```
I added a reference to an issue (see other comments), dropped the date and time because they are recorded by the commit, and added an explanation as to how this commit was generated.

Reply:
Done